src/views/sign-up/SignUp.jsx

```diff
@@ -0,0 +1,465 @@
+	const [password, setPassword] = useState({
+		value: '',
+		valid: false,
+		focused: false,
+	});
+
+	function handleOnChange(e) {
+		e.target.name === 'firstName' &&
+			changeHandler(
+				e,
+				{
+					firstName,
+					lastName,
+					userName,
+					address,
+					telephone,
+					email,
+					password,
+				},
+				validateEmptyField,
+				setFormDataValid,
+				setFirstName,
+			);
+
+		e.target.name === 'lastName' &&
+			changeHandler(
+				e,
+				{
+					firstName,
+					lastName,
+					userName,
+					address,
+					telephone,
+					email,
+					password,
+				},
+				validateEmptyField,
+				setFormDataValid,
+				setLastName,
+			);
+
+		e.target.name === 'email' &&
+			changeHandler(
+				e,
+				{
+					firstName,
+					lastName,
+					userName,
+					address,
+					telephone,
+					email,
+					password,
+				},
+				ValidateEmail,
+				setFormDataValid,
+				setEmail,
+			);
+
+		e.target.name === 'address' &&
+			changeHandler(
+				e,
+				{
+					firstName,
+					lastName,
+					userName,
+					address,
+					telephone,
+					email,
+					password,
+				},
+				validateEmptyField,
+				setFormDataValid,
+				setAddress,
+			);
+
+		e.target.name === 'userName' &&
+			changeHandler(
+				e,
+				{
+					firstName,
+					lastName,
+					userName,
+					address,
+					telephone,
+					email,
+					password,
+				},
+				validateEmptyField,
+				setFormDataValid,
+				setuserName,
+			);
+
+		e.target.name === 'telephone' &&
+			changeHandler(
+				e,
+				{
+					firstName,
+					lastName,
+					userName,
+					address,
+					telephone,
+					email,
+					password,
+				},
+				validateEmptyField,
+				setFormDataValid,
+				settelephone,
+			);
+		e.target.name === 'password' &&
+			changeHandler(
+				e,
+				{
+					firstName,
+					lastName,
+					userName,
+					address,
+					telephone,
+					email,
+					password,
+				},
+				validatePassword,
+				setFormDataValid,
+				setPassword,
+			);
+	}
+
+	function handleOnFocus(e) {
+		e.target.name === 'firstName' && focusHandler(setFirstName, firstName);
+		e.target.name === 'lastName' && focusHandler(setLastName, lastName);
+		e.target.name === 'address' && focusHandler(setAddress, address);
+		e.target.name === 'email' && focusHandler(setEmail, email);
+		e.target.name === 'userName' && focusHandler(setuserName, userName);
+		e.target.name === 'telephone' && focusHandler(settelephone, telephone);
+		e.target.name === 'password' && focusHandler(setPassword, password);
+	}
+
+	async function handleSubmit(e) {
+		try {
+			e.preventDefault();
+			const userData = {
+				firstName,
+				lastName,
+				userName,
+				address,
+				telephone,
+				email,
+				password,
+			};
+
+			const reqBody = Object.entries(userData).reduce((acc, curr) => {
+				return {
+					...acc,
+					[curr[0]]: curr[1].value,
+				};
+			}, {});
+
+			setErrorMsg('');
+			setCreatinAcc(true);
+
+			const res = await axios.post(
+				'https://e-comm-team-amigos-bn-project.onrender.com/user/create',
```

Comment:
I am  suggesting you to use ENV variables to handle base url for example `${BASE_URL}/user/create` this will make url customized and dynamic for any changes which might happen to the url.